### ISSUE TITLE
Request: Hidden tasks

### ISSUE THREAD

**Issue by Ace2213** [humhub/humhub-modules-tasks#101]
Visibility option: Only me. For truly private tasks. This makes sense to have in profiles at least, for personal goals/projects that you don't want others (even friends) to know about.

**Comment by buddh4** [comment]
The latest version allows you to install the module on profile level.

**Comment by Ace2213** [comment]
Yes, but as far as I can tell, profile tasks by default can be seen by friends. Checking "Public" makes them visible to anyone who can see your profile. 

I understand the purpose is to collaborate, but adding this feature would make it possible to use the module also for personal tasks that you don't want anyone to know about.

**Comment by Ace2213** [comment]
Temporary workaround to issue #102:

I achieved public access to the tasks profile tab by adding `User::USERGROUP_FRIEND`, `User::USERGROUP_USER` and `User::USERGROUP_GUEST` to [ChecklistController](https://github.com/humhub/humhub-modules-tasks/blob/master/controllers/ChecklistController.php), [ListController](https://github.com/humhub/humhub-modules-tasks/blob/master/controllers/ListController.php) and [TaskController](https://github.com/humhub/humhub-modules-tasks/blob/master/controllers/TaskController.php).

This means the profile tasklists are always public. I think this should be default behavior. Additionally, each task should have a permission choice of Public, Friends or Hidden/Only Me. This way, Public tasks can be seen in your tasklist by non-friends.
